daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -416,20 +418,20 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
     private def decodeLocation(lfExpr: PLF.Expr, definition: String): Option[Location] =
       if (lfExpr.hasLocation && lfExpr.getLocation.hasRange) {
         val loc = lfExpr.getLocation
-        val (pkgId, module) =
+        val optModuleRef =
           if (loc.hasModule)
-            decodeModuleRef(loc.getModule)
+            Some(decodeModuleRef(loc.getModule))
           else
-            (packageId, moduleName)
-
-        val range = loc.getRange
-        Some(
+            optModuleName.map((packageId, _))
+        optModuleRef.map(moduleRef => {
```

Comment:
```suggestion
        optModuleRef.map{ case (pkgId, module) => 
```

Reply:
Ah, that's how it's done. I was trying to do that, but couldn't convince the type checker. Apparently, you can't the argument to `map` in parentheses `(...)` if it's already wrapped in curly braces `{...}`. Thanks.